### PR TITLE
simple mapvote

### PR DESCRIPTION
**Pull request by bravo2zero**
- players can change their vote while vote is not finished
- configurable auto start when round begins
- dedicated layer list
- works with v3.0.1
- docs updated

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -827,6 +827,43 @@ Grafana:
            <pre><code>300000</code></pre></li></ul>
         </details>
 
+<details>
```
Comment (werewolfboy13) [inline README.md]: This will get over written. Needs to be added to the template instead, unless it does not do that anymore.

Comment (Thomas-Smyth) [reply README.md]: This is the bit that is generated from the plugin's options specification, so it won't get overwritten.

File: .gitignore [hunk 2]
```diff
@@ -12,3 +12,4 @@ yarn.lock
 # IDEs
 .idea/
 .vs/
+.husky/
```
Comment (Thomas-Smyth) [inline .gitignore]: This isn't an IDE and is required, so it shouldn't be added to a `.gitignore` file.

**Comment by werewolfboy13** [comment]
Has this been tested on a live server?

**Comment by luqero** [comment]
How to make it work? It say that layers are required. How to add layers parameter to config file? Like this seems to not work:
(..)
 "layersHistorySize": 3,
 "layers": "Anvil_AAS_v1,Anvil_AAS_v2,Anvil_Invasion_v1,Anvil_Invasion_v2,(..)" 

EDIT:
Got it working with this config:
 "layers": [
"Anvil_AAS_v1",
"Anvil_AAS_v2",
"Anvil_Invasion_v1",
"Anvil_Invasion_v2",
"Anvil_RAAS_v1",
"Anvil_RAAS_v2",
"Anvil_RAAS_v3",
"Anvil_RAAS_v4"
]

**Comment by bravo2zero** [comment]
@luqero correct, that is just an array of layer ids that you want to have included in voting. i've added full layer list as default and updated docs, so that one could tweak the list as he sees fit.

**Comment by bravo2zero** [comment]
> Has this been tested on a live server?

yeap

**Comment by bravo2zero** [comment]
also tested on live server with the latest v3 update

**Comment by lbzepoqo** [comment]
tested on a live server, there were multiple instances that it didn't read the votes when using the automatic vote after start of match and had to restart the bot for it to work

This is my config

`     "plugin": "MapVote",
      "enabled": true,
      "voteDurationSeconds": 120,
      "minimumVotes": 1,
      "autoStartMapVoteSeconds": 120,
      "autoRepeatVoteBroadcastSeconds": 30,
      "layersHistorySize": 3,
      "layers": 
`

**Comment by speccycy** [comment]
MpaVote doesn't work on Mod Server

**Comment by bravo2zero** [comment]
> MpaVote doesn't work on Mod Server

this is most likely due to different Layer path logged, if you have your full log lines with "Bringing World..." when starting new layer, we might figure out regex fix

**Comment by bravo2zero** [comment]
> MapVote

was it working when you start vote manually with chat command? what was the logged lines for "MapVote" logger?

**Comment by bravo2zero** [comment]
> tested on a live server, there were multiple instances that it didn't read the votes when using the automatic vote after start of match and had to restart the bot for it to work
> 
> This is my config
> 
> ` "plugin": "MapVote", "enabled": true, "voteDurationSeconds": 120, "minimumVotes": 1, "autoStartMapVoteSeconds": 120, "autoRepeatVoteBroadcastSeconds": 30, "layersHistorySize": 3, "layers": `

was it working when you start vote manually with chat command? what was the logged lines for "MapVote" logger?

**Comment by werewolfboy13** [comment]
@bravo2zero doing clean up here. Can you resolve the conflicts. Also is this stable? Last I heard it wasn't stable.

File: .gitignore (hunk 2 above)
Comment (werewolfboy13) [reply .gitignore]: Still needs resolving.

**Review by werewolfboy13 — CHANGES_REQUESTED** [review]
.gitignore needs resolving

**Pull request closed without merging**